src/solver/optimisation/adequacy_patch.cpp

```diff
@@ -222,9 +222,9 @@ std::pair<double, double> calculateAreaFlowBalance(PROBLEME_HEBDO* ProblemeHebdo
     return std::make_pair(netPositionInit, densNew);
 }
 
-void addArray(std::vector<double>& A, double* B, int num)
+void addArray(std::vector<double>& A, double* B)
 {
-    for (uint i = 0; i < num; ++i)
+    for (uint i = 0; i < A.size(); ++i)
         A[i] += B[i];
```

Comment:
if you remove the num and replace it with A.size(), there should be some additional check that B.size >= A.size, or B.size = A.size. 
Otherwise, B[i] could be out of index error

Reply:
I understand your point. But this will be a little tricky. 
A is of type std::vector<double> so it is easy to find it's size just by A.size().
B is of type double* (let's say array). Arrays decay to pointers when passed as function parameters so even sizeof(B)/sizeof(B[0]) will just return 1 all the time. 
We will just have to trust it I guess.